src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -441,14 +472,46 @@ bool LFUDAPolicy::erase_dirty_object(const DoutPrefixProvider* dpp, const std::s
     return false;
   }
 
-  object_heap.erase(p->second->handle);
+  object_heap.erase(p->second.first->handle);
   o_entries_map.erase(p);
-  delete p->second;
-  p->second = nullptr;
+  delete p->second.first;
+  p->second.first = nullptr;
+  p->second.second = State::DONE;
```

Comment:
Is this valid? You are erasing the map entry and then setting the state to DONE. Shouldn't the state be set to DONE first and then erased?

Reply:
Also you need to call cond.notify_one() to wake the condition variable.